### PR TITLE
Fix a bug and a few formatting errors

### PR DESCRIPTION
**Pull request by LukasKalbertodt**
Thanks for this RFC! I noted a few small things which I fixed in this PR. I hope you agree with all changes :)

### CODE REVIEW

File: 0000-nonlexical-lifetimes.md [hunk 1]
```diff
@@ -1575,8 +1579,8 @@ Another example, this time using a `match`:
 ```rust
 fn main() {
     let mut x = Some(3);
-    match &mut x {
-        Some(i) => {
+    match *&mut x {
+        Some(ref mut i) => {
```
Comment (nikomatsakis) [inline 0000-nonlexical-lifetimes.md]: I can't remember, but I think I may have been intending to use the notation introduced by https://github.com/rust-lang/rfcs/pull/2005 here (which is not, admittedly, yet implemented).

Comment (nikomatsakis) [reply 0000-nonlexical-lifetimes.md]: Nonetheless, I think I'd prefer to keep it the way it was.

File: 0000-nonlexical-lifetimes.md [hunk 2]
```diff
@@ -1593,9 +1597,9 @@ The error might be:
 error[E0506]: cannot write to `x` while borrowed
  --> <anon>:4:5
    |
- 3 |     match &mut x {
-   |           ------ (mutable) borrow of `x` occurs here
- 4 |         Some(i) => {
+ 3 |     match *&mut x {
+   |            ------ (mutable) borrow of `x` occurs here
+ 4 |         Some(ref mut i) => {
```
Comment (nikomatsakis) [inline 0000-nonlexical-lifetimes.md]: Similarly here

File: 0000-nonlexical-lifetimes.md (hunk 1 above)
Comment (LukasKalbertodt) [reply 0000-nonlexical-lifetimes.md]: But your version [doesn't compile](https://play.rust-lang.org/?gist=006ac19b2c85e558bff35c6b3b2a80c7&version=stable). That's the only reason I fixed it. It's also possible to solve like this:

```rust
match &mut x {
    &mut Some(ref mut i) => { ... }
    &mut None => { ... }
}       
```
But I think that this style is discouraged.

Or am I misunderstanding your example and it is supposed to fail compiling due to that error?

**Comment by LukasKalbertodt** [comment]
I rebased and fixed the conflicts. Should be up to date now again.

**Comment by LukasKalbertodt** [comment]
So I guess I can close this now, yes? :crying_cat_face: 

---

Or shall I rebase and open a PR to your `rfcs` repo? I can leave out the formatting changes (although your current formatting doesn't conform with the official style in some snippets). But apart from that, I think I only fixed typos and small mistakes in snippets, which are still in the version you "published" to the official rfcs repo. Or do you disagree?

You can also just take my changes and manually apply them. I don't want fame, but just to fix bugs :stuck_out_tongue:

File: 0000-nonlexical-lifetimes.md (hunk 1 above)
Comment (nikomatsakis) [reply 0000-nonlexical-lifetimes.md]: It will compile when rust-lang/rfcs#2005 is implemented, is the point.

**Comment by nikomatsakis** [comment]
@LukasKalbertodt I'm still keeping this repo as the "official" text of the NLL RFC for now. I'll merge in your changes, or at least most of them, in a bit. =)

File: 0000-nonlexical-lifetimes.md (hunk 1 above)
Comment (LukasKalbertodt) [reply 0000-nonlexical-lifetimes.md]: Oh, I wasn't aware of that, sorry! I'll remove those changes.

**Comment by LukasKalbertodt** [comment]
I rebased and tweaked my changes. I didn't include all the changes from my original PR. I hope these changes are all uncontroversial :3

**Comment by nikomatsakis** [comment]
merged, sorry for the delay =)

**Pull request merged**